components/retro-go/rg_display.c

```diff
@@ -36,6 +39,74 @@ static const char *SETTING_CUSTOM_ZOOM = "DispCustomZoom";
 #include "drivers/display/dummy.h"
 #endif
 
+// TODO : make it more user-friendly -> instead of specifying a rect and a surface
+//        just specify the corner and the dimensions (width and height)
+rg_surface_t *rg_display_init_osd(rg_corner_t corner, int width, int height, bool has_transparency)
+{
+    // clipping if osd > display
+    width = RG_MIN(width, display.screen.width);
+    height = RG_MIN(height, display.screen.height);
+
+    osd.has_transparency = has_transparency;
+
+    int left, top;
+    switch (corner)
+    {
+    case CORNER_TOP_LEFT:
+        left = 0;
+        top = 0;
+        break;
+    case CORNER_TOP_RIGHT:
+        left = display.screen.width - width;
+        top = 0;
+        break;
+    case CORNER_BOTTOM_LEFT:
+        left = 0;
+        top = display.screen.height - height;
+        break;
+    case CORNER_BOTTOM_RIGHT:
+        left = display.screen.width - width;
+        top = display.screen.height - height;
+        break;
+    default:
+        left = 0;
+        top = 0;
+        break;
+    }
+
+    osd.surface = rg_surface_create(width, height, RG_PIXEL_565_LE, MEM_SLOW);
+
+    osd.left = left;
+    osd.top = top;
+
+    if (has_transparency)
+    {   // we fill the background with the "transparent color"
+        uint16_t *buffer = osd.surface->data; // Treat the buffer as 16-bit values
+        for (int i = 0; i < osd.surface->width * osd.surface->height; i++)
+            buffer[i] = C_TRANSPARENT; // Assign the C_TRANSPARENT color directly to the background
```

Comment:
I have finally implemented rg_surface_fill, you should now be able to do `rg_surface_fill(osd.surface, NULL, C_TRANSPARENT);`

Reply:
Although I'm not able to test my own code right now so maybe best don't waste your time...